How-to-find-the-commit-which-introduced-a-bug.md

```diff
@@ -0,0 +1,186 @@
+## Finding Bad Commits with the help of `git blame` and `git bisect`
+
+#### Before Using both these commands:
+1. **Make sure to checkout to the latest commit and see if the error exists.**
+    * **If the error is not reproducible on the latest commit:** we don’t need to do anything! You can post video proof on the issue thread and request that a maintainer close the issue.
+    * **Otherwise**: continue with the steps below.
+2. **Depending on how far back you want to go, checkout one of the following:**
+    * **This commit from Dec 2022 &lt;[9a334e9bde1d3d10e3b69dcd461d3e649733b0c0](https://github.com/oppia/oppia/commit/9a334e9bde1d3d10e3b69dcd461d3e649733b0c0)>, which is when the Python version was last updated – though this might not work if you are using Docker.**
+    * **This commit from Aug 2023 &lt;<code>[dc333e4](https://github.com/oppia/oppia/commit/dc333e4e25dc72e22910cb6f8ef32ae652a29dad)</code>>, which is when Docker was first usable.</strong>
+    * <strong>The latest release commit (you can find the hash and release date [here](https://github.com/oppia/oppia/releases)), if you want to see whether the error happened since the last release;</strong>
+3. <strong>If the issue is <span style="text-decoration:underline;">not</span> reproducible on that commit then go ahead with the approach provided below.</strong>
+4. <strong>If the issue is reproducible on that commit, report that info on the issue thread or group chat as appropriate. Include a link to the commit and mention the month/year it was made.</strong>
+
+<!-- <span style="text-decoration:underline; font-size: 20px;">Git Blame</span> -->
```

Comment:
Please drop all html constructs from this file.

Reply:
Done